service/eventer/github/github.go

```diff
@@ -111,15 +111,18 @@ func (e *Eventer) FetchContinuously(projects []string, environment string) (<-ch
 		// TODO this map just grows which means we fill the programm with memory.
 		etagMap := make(map[string]string)
 
-		for c := ticker.C; ; <-c {
-			for _, p := range projects {
-				deployments, err := e.fetchNewDeploymentEvents(p, environment, etagMap)
-				if err != nil {
-					e.logger.Log("error", "could not fetch deployment events", "message", err.Error())
-				}
-
-				for _, d := range deployments {
-					deploymentEventChannel <- d.DeploymentEvent(p)
+		for {
+			select {
+			case <-ticker.C:
+				for _, p := range projects {
+					deployments, err := e.fetchNewDeploymentEvents(p, environment, etagMap)
+					if err != nil {
+						e.logger.Log("error", "could not fetch deployment events", "message", err.Error())
+					}
+
+					for _, d := range deployments {
+						deploymentEventChannel <- d.DeploymentEvent(p)
+					}
```

Comment:
A bit boyscouting. Initially I tried to added some shutdown mechanism to the informer and eventer to properly teardown tests. I had something but decided against it now to make progress and not get lost it making things nice. Turned out the shutdown magic is not necessary for the testing so I removed some code again. This code here is left over because I think it is more extendable that way. Now a context's done channel can easily be hooked into. Something we can improve later on.